src/Microsoft.IdentityModel.Abstractions/TelemetryEventDetails.cs

```diff
@@ -0,0 +1,127 @@
+            {
+                return (IReadOnlyDictionary<string, object>)PropertyValues;
+            }
+        }
+
+        /// <summary>
+        /// Sets a property on the event details.
+        /// </summary>
+        /// <param name="key">Property key.</param>
+        /// <param name="value">Property value.</param>
+        /// <exception cref="ArgumentNullException">'key' is null.</exception>
+        public virtual void SetProperty(
+            string key,
+            string value)
+        {
+            SetPropertyCore(key, value);
+        }
+
+        /// <summary>
+        /// Sets a property on the event details.
+        /// </summary>
+        /// <param name="key">Property key.</param>
+        /// <param name="value">Property value.</param>
+        /// <exception cref="ArgumentNullException">'key' is null.</exception>
+        public virtual void SetProperty(
+            string key,
+            long value)
+        {
+            SetPropertyCore(key, value);
+        }
+
+        /// <summary>
+        /// Sets a property on the event details.
+        /// </summary>
+        /// <param name="key">Property key.</param>
+        /// <param name="value">Property value.</param>
+        /// <exception cref="ArgumentNullException">'key' is null.</exception>
+        public virtual void SetProperty(
+            string key,
+            bool value)
+        {
+            SetPropertyCore(key, value);
+        }
+
+        /// <summary>
+        /// Sets a property on the event details.
+        /// </summary>
+        /// <param name="key">Property key.</param>
+        /// <param name="value">Property value.</param>
+        /// <exception cref="ArgumentNullException">'key' is null.</exception>
+        public virtual void SetProperty(
+            string key,
+            DateTime value)
+        {
+            SetPropertyCore(key, value);
+        }
+
+        /// <summary>
+        /// Sets a property on the event details.
+        /// </summary>
+        /// <param name="key">Property key.</param>
+        /// <param name="value">Property value.</param>
+        /// <exception cref="ArgumentNullException">'key' is null.</exception>
+        public virtual void SetProperty(
+            string key,
+            double value)
+        {
+            SetPropertyCore(key, value);
+        }
+
+        /// <summary>
+        /// Sets a property on the event details.
+        /// </summary>
+        /// <param name="key">Property key.</param>
+        /// <param name="value">Property value.</param>
+        /// <exception cref="ArgumentNullException">'key' is null.</exception>
+        public virtual void SetProperty(
+            string key,
+            Guid value)
+        {
+            SetPropertyCore(key, value);
+        }
+
+        private void SetPropertyCore(
+            string key,
+            object value)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
```

Comment:
Will we be piping this through our Logger?
Should we use our LogHelper?

Reply:
GoodQ: It's not related to the logger. Thought it may be injected in a similar way down into Wilson at least for now when we want to integrate telemetry for e.g. crypto operations.